source/lib/src/rocm/CMakeLists.txt

```diff
@@ -1,18 +1,21 @@
 # required cmake version
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.21)
```

Comment:
Do we use something new in this CMAKE file? I am afraid of cmake version on some platform.

Reply:
I see CMAKE_HIP_STANDARD is new in 3.21.

Reply:
I see HIP was added to CMake just one year ago: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/6121. Did ROCm itself provide it?

Reply:
Wow, `ROCm` is supported by cmake. I guess we don't need to maintain FindROCM.cmake ourselves

Reply:
Indeed it just provides HIP language... But not FindHIP.

Reply:
I guess `CMAKE_HIP_STANDARD` does not work here. ROCm's CMake module does not provide such a variable. https://github.com/ROCm-Developer-Tools/HIP/blob/develop/cmake/FindHIP.cmake

Reply:
Yeah, there isn't a macro like `CUDA_LIBRARY`. But this [macro](https://github.com/Kitware/CMake/blob/c55a41c9b2fea1617ad2a0a50760e4f8b6e42a47/Modules/CMakeHIPCompiler.cmake.in#L20) maybe help us to find the `ROCM_ROOT`

Reply:
It looks working!

Reply:
```cmake
cmake_minimum_required(VERSION 3.21)
include(CMakeDetermineHIPCompiler)
set(ROCM_search_PATHS ${CMAKE_HIP_COMPILER_ROCM_ROOT})
```

The cmake native module `CMakeDetermineHIPCompiler` is used to find CMAKE_HIP_COMPILER_ROCM_ROOT (can be overrides by user).